### PR TITLE
pre-commit hook for secret detection

### PR DESCRIPTION
**Pull request by sharon-wang**
### Intent
- Addresses https://github.com/rstudio/rstudio-pro/issues/5605

### Approach
- Uses https://github.com/Yelp/detect-secrets to generate a baseline secrets file, mark false positives and run a hook to detect secrets
- Added secret detection functionality to existing pre-commit hook
- Implemented a wrapper script `git_hooks/secrets/run-detect-secrets` around the `detect-secrets` tool, to make manual calls to detect secrets simpler
- Added docs on how to install the pre-commit hook, update the baseline with new secrets, mark false positives, etc.
- Added a manual test to confirm that GitHub and AWS tokens are detected properly. See `git_hooks/secrets/test/README.md`.

It's possible we'll need to modify the entropies for `Base64HighEntropyString` and `HexHighEntropyString` if the false positives are occurring too often, as we commit new code.

A separate PR for pro will be opened once these changes are merged over.

### Automated Tests
No automated tests. Just the one manual test mentioned above.

### QA Notes
For anyone following the docs `git_hooks/secrets/README.md`, it should lead them to successfully set up the pre-commit hook and install `detect-secrets`. Then, when attempting to commit some code with something that looks like a secret, the commit should fail.

### Documentation
Just dev docs `git_hooks/README.md`, `git_hooks/secrets/README.md` and `git_hooks/secrets/test/README.md`.

### Checklist

~- [ ] If this PR adds a new feature, or fixes a bug in a previously released version, it includes an entry in `NEWS.md`~ 
~- [ ] If this PR adds or changes UI, the updated UI meets [accessibility standards](https://github.com/rstudio/rstudio/wiki/Accessibility)~
- [x] A reviewer is assigned to this PR (if unsure who to assign, check Area Owners list)
- [x] This PR passes all local unit tests

### CODE REVIEW

**Review by cm421 — APPROVED** [review]
LGTM!

File: git_hooks/set-up-git-hooks [hunk 1]
```diff
@@ -11,14 +11,47 @@ for hook in "$hooks_dir"/*
...
+
+# check if detect-secrets is installed and prompt to install if it is not
+# if the user opts out of installing detect-secrets here, they will be prompted again
+# when they try to commit, assuming they have the secret scanning hook enabled
+if ! command -v detect-secrets &> /dev/null
+then
+    read -p $'\t\033[1;95mdetect-secrets\033[0m is not installed. Would you like to install it? (y/n): ' -r
+    if [[ $REPLY =~ ^[Yy]$ ]]
+    then
+        read -p $'\tWould you like to install using \033[1;95mpip\033[0m (p) or \033[1;95mbrew\033[0m (b)?: ' -r
```
Comment (GCRev) [inline git_hooks/set-up-git-hooks:40]: Didn't want to also use `{color_highlight}` and `{color_none}` here?

**Review by GCRev — APPROVED** [review]
I think we're good to go.

File: git_hooks/set-up-git-hooks (hunk 1 above)
Comment (sharon-wang) [reply git_hooks/set-up-git-hooks:40]: I should've added a comment to explain, but I couldn't get the colour variables `{color_highlight}` and `{color_none}` to work with the `read` command using the `-p` prompt.

I could change these to use `echo` to display the prompt and then `read` in the input separately though

Comment (GCRev) [reply git_hooks/set-up-git-hooks:40]: Then it's probably not worth agonizing over.

**Pull request merged**